pygmt/src/coupe.py

```diff
@@ -0,0 +1,416 @@
+"""
+coupe - Plot cross-sections of focal mechanisms.
+"""
+
+import numpy as np
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    build_arg_list,
+    data_kind,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.src._common import _FocalMechanismConvention
```

Comment:
You can directly import these functions and use them in `Figure.coupe`. Then you can remove lines from 32 to 120:
```suggestion
from pygmt.src.meca import _get_focal_convention, _preprocess_spec, _auto_offset
```